distributed/batched.py

```diff
@@ -94,9 +95,24 @@ def _background_send(self):
             self.batch_count += 1
             self.next_deadline = time() + self.interval
             try:
-                nbytes = yield self.comm.write(
-                    payload, serializers=self.serializers, on_error="raise"
-                )
+                # NOTE: Since `BatchedSend` doesn't have a handle on the running
+                # `_background_send` coroutine, the only thing with a reference to this
+                # coroutine is the event loop itself. If the event loop stops while
```

Comment:
the event loop doesn't even keep a reference to the coro, necessarily

it's  `loop._ready`, `loop._scheduled`, `loop._selector` -> Future.callback(self) ->  Future._callbacks -> `Task.__step(self)`, -> `self._coro`

and when you call `loop.close()` it wipes out the ready, scheduled and selector
however you can also create a Future wait on it in a Task and then lose the Future the coro goes with it

Reply:
actually on second thought maybe it is worth a change here, because maybe it explains why you can't fix this with `yield asyncio.create_task(self.comm.write(...))`. Or maybe It just depends on if the hammer falls before or after `asyncio.get_running_loop()` starts failing